src/components/dashboard/GovernanceSection/GovernanceSection.tsx

```diff
@@ -85,73 +82,76 @@ const GovernanceSection = () => {
-            Use your SAFE tokens to vote on important proposals or participate in forum discussions.
-          </Typography>
-        </div>
-      </AccordionSummary>
-
-      <AccordionDetails sx={({ spacing }) => ({ padding: `0 ${spacing(3)}` })}>
-        {governanceApp || fetchingSafeGovernanceApp ? (
-          <WidgetBody>
-            <Card className={css.widgetWrapper}>
-              {governanceApp && !safeLoading ? (
-                <MiniAppFrame app={governanceApp} title="Safe Governance" />
-              ) : (
-                <Box
-                  className={css.widgetWrapper}
-                  display="flex"
-                  alignItems="center"
-                  justifyContent="center"
-                  textAlign="center"
-                >
-                  <Typography variant="h1" color="text.secondary">
-                    Loading section...
-                  </Typography>
-                </Box>
-              )}
-            </Card>
-          </WidgetBody>
-        ) : (
-          <Alert severity="warning" elevation={3}>
-            There was an error fetching the Governance section. Please reload the page.
-          </Alert>
-        )}
-      </AccordionDetails>
-    </Accordion>
+    <>
+      {governanceApp || fetchingSafeGovernanceApp ? (
+        <WidgetBody>
+          <Card className={css.widgetWrapper}>
+            {governanceApp && !safeLoading ? (
+              <MiniAppFrame app={governanceApp} title="Safe Governance" />
+            ) : (
+              <Box
+                className={css.widgetWrapper}
+                display="flex"
+                alignItems="center"
+                justifyContent="center"
+                textAlign="center"
+              >
+                <Typography variant="h1" color="text.secondary">
+                  Loading section...
+                </Typography>
+              </Box>
+            )}
+          </Card>
+        </WidgetBody>
+      ) : (
+        <WidgetLoadErrorFallback />
+      )}
+    </>
   )
 }
 
-// Prevent `GovernanceSection` hooks from needlessly being called
-const GovernanceSectionWrapper = () => {
-  const chainId = useChainId()
-  const [isLayoutReady, setIsLayoutReady] = useState(false)
+const LazyGovernanceSection = () => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [hasScrolled, setHasScrolled] = useState(false)
+
+  const onVisible = useCallback(() => {
+    setIsVisible(true)
+  }, [])
 
   useEffect(() => {
-    setIsLayoutReady(true)
+    const handleScroll = () => {
+      if (window.scrollY > 0) {
+        setHasScrolled(true)
+        window.removeEventListener('scroll', handleScroll)
+      }
+    }
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
   }, [])
 
-  if (!isLayoutReady) {
-    return null
-  }
+  return (
+    <>
+      {hasScrolled && <InfiniteScroll onLoadMore={onVisible} />}
```

Comment:
Why do you need hasScrolled? Can't we just render the infinitescroll all the time?

Reply:
The page is small in the beginning and this block is always in the viewport initially.